### DISCUSSION TITLE
there is no Joint class in assimp? so how to animate the fbx model with pose of Joints by manually set every frame?

### DISCUSSION THREAD

**Discussion by opengpu** [assimp/assimp#4983] [Q&A]
there is no Joint class in assimp? so how to animate the fbx model with pose of Joints?
Thanks

**Comment by kimkulling** [comment]
Right, you need to use the aiBones. The animations in Assimp bringing some legacy with them. Let me start:

- aiScene stores the meshes and the node for your mesh
- Each mesh stores an array with bones
  - A bone stores its name, the indices of vertex weights in the mesh and the offset matrix which will be used to convert the transformation from model-space to bone-space
- The animation channels are stored in the datatype aiAnimation, which holds next to the tick and time all the animation channels. 

So when doing any kind of bone animation you need to get the Bones from your meshes. The link to the related bones are made by their name.

You can follow this tutorial to see how to set this up: [Assimp Bone Animation](https://learnopengl.com/Guest-Articles/2020/Skeletal-Animation)
